subprojects/tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinLibSources.kt

```diff
@@ -109,7 +109,8 @@ fun resolveSourcesUsing(dependencyHandler: DependencyHandler, query: ArtifactRes
             .resolvedComponents
             .flatMap { it.getArtifacts(SourcesArtifact::class.java) }
             .filterIsInstance<ResolvedArtifactResult>()
-            .map { it.file })
+            .map { it.file }
+            .sorted()) // TODO remove sorting once https://github.com/gradle/gradle/issues/5507 is fixed
```

Comment:
Why do you sort by `it.name` above and just by the natural `File` sorting behavior here?

Reply:
Above is sorting child directories of a given directory.
This is a temporary workaround sorting a set of files coming out in unpredictable order from an `ArtifactResolutionQuery`.